modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1579,8 +1579,15 @@ void ONNXImporter::lstm_extractConsts(LayerParams& layerParams, const opencv_onn
         Mat blob;
         if (idx < lstm_proto.input_size() && !lstm_proto.input(idx).empty())
         {
-            blob = getBlob(lstm_proto, idx);
-            CV_Assert(shape(blob) == blobShape);
+            if ((idx == 5 or idx == 6) and (constBlobs.find(lstm_proto.input(idx)) == constBlobs.end()))
```

Comment:
1. Could you please use c++ operators?
2. We're not using this blob at all, but we allocate memory for it. Not sure it's the right approach.

Reply:
1. Done 
2. Solution provided follows legacy initialisation style. If blobs are not found in constBlobs, they are initialised as 0's and are actually used in `forward` of LSTM layer

Reply:
2) If we have dynamic hx input, we allocate a new mat with zeros for hx. And then in LSTM forward we get hx as input and don't look at blobs. So, why did we allocate it?